projects/js-packages/publicize-components/src/components/services/tests/service-connection-info.test.js

```diff
@@ -0,0 +1,95 @@
+} ) );
+
+jest.mock( '../../connection-management/connection-name', () => ( {
+	ConnectionName: ( { connection } ) => <div>{ connection.display_name }</div>,
+} ) );
+jest.mock( '../../connection-management/connection-status', () => ( {
+	ConnectionStatus: ( { connection } ) => <div>Status: { connection.status }</div>,
+} ) );
+jest.mock( '../../connection-management/disconnect', () => ( {
+	Disconnect: ( { connection } ) => <button>Disconnect { connection.display_name }</button>,
+} ) );
+jest.mock( '../../connection-management/mark-as-shared', () => ( {
+	MarkAsShared: () => <button>Mark as Shared</button>,
+} ) );
+
+describe( 'ServiceConnectionInfo', () => {
+	const connection = {
+		profile_picture: 'https://example.com/profile.jpg',
+		display_name: 'Example User',
+		status: 'connected',
+		can_disconnect: true,
+	};
+
+	const service = {
+		icon: () => <svg aria-label="test-svg"></svg>,
+	};
+
+	const renderComponent = ( connOverrides = {}, serviceOverrides = {} ) => {
+		render(
+			<ServiceConnectionInfo
+				connection={ { ...connection, ...connOverrides } }
+				service={ { ...service, ...serviceOverrides } }
+			/>
+		);
+	};
+
+	test( 'renders profile picture if available', () => {
+		renderComponent();
+		const profilePic = screen.getByAltText( 'Example User' );
+		expect( profilePic ).toBeInTheDocument();
+		expect( profilePic ).toHaveAttribute( 'src', 'https://example.com/profile.jpg' );
+	} );
+
+	test( 'renders service icon if profile picture is not available', () => {
+		renderComponent( { profile_picture: null } );
+		const serviceIcon = screen.getByLabelText( 'test-svg' );
+		expect( serviceIcon ).toBeInTheDocument();
+	} );
+
+	test( 'displays ConnectionName', () => {
+		renderComponent();
+		expect( screen.getByText( 'Example User' ) ).toBeInTheDocument();
+	} );
+
+	test( 'displays ConnectionStatus if status is broken', () => {
+		renderComponent( { status: 'broken' } );
+		expect( screen.getByText( 'Status: broken' ) ).toBeInTheDocument();
+	} );
+
+	test( 'displays MarkAsShared button if connection can be disconnected', () => {
+		renderComponent();
+		expect( screen.getByText( 'Mark as Shared' ) ).toBeInTheDocument();
+	} );
+
+	test( 'displays disconnect button', () => {
+		renderComponent();
+		expect( screen.getByText( 'Disconnect Example User' ) ).toBeInTheDocument();
+	} );
+
+	test( 'displays description if connection cannot be disconnected', () => {
+		renderComponent( { can_disconnect: false } );
+		expect(
+			screen.getByText( 'This connection is added by a site administrator.' )
+		).toBeInTheDocument();
+	} );
+
+	test( 'displays IconTooltip when hovering over MarkAsShared button', async () => {
+		renderComponent();
+		const markAsSharedButton = screen.getByText( 'Mark as Shared' );
+		await userEvent.hover( markAsSharedButton );
+		expect(
+			screen.getByText(
+				'If enabled, the connection will be available to all administrators, editors, and authors.'
+			)
+		).toBeInTheDocument();
+	} );
```

Comment:
Tooltip is displayed on click not on hover.

Reply:
Good point! This is interesting, because it did pass! That means that it's visible in the Dom even without clicking

Reply:
Yeah, that means we need to test for its visibility, i.e. it should not be visible on hover but should be visible on click.

Reply:
I made it so we test that it's not visible without clicking it. Testing that it shows up on click would be testing the IconTooltip component, which I think we do not need to do on second thought [94a115d](https://github.com/Automattic/jetpack/pull/37582/commits/94a115d636cbbb982f34538cb026c38d9333e77e)